### DISCUSSION TITLE
Add a black background upto screen dimensions

### DISCUSSION THREAD

**Discussion by lamyergeier** [ImageMagick/ImageMagick#7532] [Help] [answered]
### ImageMagick version

7.1.1-33 

### Operating system, version and so on

Fedora 40

### Description

# I did the following

## Original.jpg

![Current gen](https://github.com/user-attachments/assets/a7ee7c80-3850-4303-ae72-70be176bb88b)

## Generate black background image

In the following, x and y are screen dimension

```bash
BackgroundP="$(mktemp --suffix=.jpg)"
BackgroundColor='black'
x="2256"
y="1504"
magick -size "${x}"x"${y}" "xc:${BackgroundColor}" -flatten "${BackgroundP}"
```

## Put Original.jpg  on top  of the background to produce Output.jpg

```bash
magick composite Original.jpg $BackgroundP -gravity center Output.jpg
```

## Output.jpg

![o](https://github.com/user-attachments/assets/5b73b670-9cbc-451a-87de-d503ad0b5a69)

# Issues

1. the image turned black and white! Why did this color change happened?

# NOTE

The final dimensions of Output.jpg should be 2256 × 1504

**Comment by snibgo** [comment]
Instead of using `magick composite`, I suggest using just `magick` with a `-composite` operation. This needs the input images in the opposite order.
```
magick $BackgroundP Original.jpg -gravity center -composite Output.jpg
```
The background image has one grayscale channel, and the output of -composite` inherits this. Use `-colorspace sRGB` to give the input three channels:
```
magick $BackgroundP -colorspace sRGB Original.jpg -gravity center -composite Output.jpg
```
If you want, you can combine the two `magick` commands. This avoids writing the intermediate result:

```
BackgroundColor='black'
x="2256"
y="1504"

magick -size "${x}"x"${y}" "xc:${BackgroundColor}" -flatten -colorspace sRGB Original.jpg -gravity center -composite Output.jpg
```

**Comment by GeeMack** [comment, marked as answer]
If the task is to place an image centered on a black background of particular dimensions, I'd probably start with "-extent" like this...

    x="2256"
    y="1504"
    
    magick input.png -background black -gravity center -extent "${x}"x"${y}" result.png
